superdesk/roles/roles.py

```diff
@@ -93,11 +93,12 @@ def __send_notification(self, updates, role):
             added, removed, modified = compare_preferences(role.get('privileges', {}), updates['privileges'])
             if len(removed) > 0 or (1, 0) in modified.values():
                 push_notification('role_privileges_revoked', updated=1, role_id=str(role_id))
-            if len(added) > 0:
-                add_activity(ACTIVITY_UPDATE,
-                             'role {{role}} is granted new privileges: Please re-login.',
-                             self.datasource,
-                             notify=notified_users,
-                             role=role.get('name'))
+            if len(added) > 0 or (0, 1) in modified.values():
+                activity = add_activity(ACTIVITY_UPDATE,
+                                        'role {{role}} is granted new privileges: Please re-login.',
+                                        self.datasource,
+                                        notify=notified_users,
+                                        role=role.get('name'))
```

Comment:
Do we need 2 push notifications? If not can we do the same thing what we are doing in desks.py?